rules/rollup_json.go

```diff
@@ -0,0 +1,196 @@
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package rules
+
+import (
+	"sort"
+
+	"github.com/m3db/m3metrics/policy"
+)
+
+// RollupTargetJSON is a common json serializable rollup target. Implements Sort interface.
+type RollupTargetJSON struct {
+	Name     string          `json:"name" validate:"required"`
+	Tags     []string        `json:"tags" validate:"required"`
+	Policies []policy.Policy `json:"policies" validate:"required"`
+}
+
+// RollupRuleJSON is a common json serializable rollup rule. Implements Sort interface.
+type RollupRuleJSON struct {
+	ID                  string             `json:"id,omitempty"`
+	Name                string             `json:"name" validate:"required"`
+	Filter              string             `json:"filter" validate:"required"`
+	Targets             []RollupTargetJSON `json:"targets" validate:"required,dive,required"`
+	CutoverMillis       int64              `json:"cutoverMillis,omitempty"`
+	LastUpdatedBy       string             `json:"lastUpdatedBy"`
+	LastUpdatedAtMillis int64              `json:"lastUpdatedAtMillis"`
+}
+
+// NewRollupTargetJSON takes a RollupTargetView and returns the equivalent RollupTargetJSON.
+func NewRollupTargetJSON(t RollupTargetView) RollupTargetJSON {
+	return RollupTargetJSON(t)
+}
+
+// NewRollupRuleJSON takes a RollupRuleView and returns the equivalent RollupRuleJSON.
+func NewRollupRuleJSON(rrv *RollupRuleView) RollupRuleJSON {
+	targets := make([]RollupTargetJSON, len(rrv.Targets))
+	for i, t := range rrv.Targets {
+		targets[i] = NewRollupTargetJSON(t)
+	}
+	return RollupRuleJSON{
+		ID:                  rrv.ID,
+		Name:                rrv.Name,
+		Filter:              rrv.Filter,
+		Targets:             targets,
+		CutoverMillis:       rrv.CutoverNanos / nanosPerMilli,
+		LastUpdatedBy:       rrv.LastUpdatedBy,
+		LastUpdatedAtMillis: rrv.LastUpdatedAtNanos / nanosPerMilli,
+	}
+}
+
+// RollupTargetView returns the equivalent RollupTargetView.
+func (t RollupTargetJSON) RollupTargetView() RollupTargetView {
```

Comment:
nit: `ToRollupTargetView` might be better. Also please group the constructor for the type together with the member methods for that type so the code is more readable.